command/jsonprovider/block.go

```diff
@@ -75,3 +61,20 @@ func marshalBlock(configBlock *configschema.Block) *block {
 
 	return &ret
 }
+
+func nestingModeString(mode configschema.NestingMode) string {
+	switch mode {
+	case configschema.NestingSingle:
+		return "single"
+	case configschema.NestingGroup:
+		return "group"
```

Comment:
@mildwonkey Just to confirm - this nesting mode is still valid for blocks only, not attributes, correct? So this codepath would/should never get executed for attributes, or (worst case scenario) it would panic here at some point:

https://github.com/hashicorp/terraform/blob/783936f4c725a7e285f489453b5c7f9976912376/configs/configschema/implied_type.go#L72-L82

Reply:
correct! For some additional info:
*  my next task involves extending the largely unused `InternalValidate` function so we can validate Attribute schemas immediately and return an error early at load time if there is a problem 
* the protocol doesn't include `NestingGroup` as an option for attributes, it uses a different type for an attribute's nesting modest o limit the selection